decoder.go

```diff
@@ -183,17 +185,14 @@ func (d *Decoder) DecodeBytes() ([]byte, error) {
 	if d.offset >= len(d.p) {
 		return nil, io.ErrUnexpectedEOF
 	}
+
 	l, n, err := DecodeVarint(d.p[d.offset:])
+	err = d.validateFieldLen(l, n, err)
```

Comment:
nit, you could inline this with the error check which would be consistent with that you've done elsewhere
```suggestion
	if err = d.validateFieldLen(l, n, err); err != nil {
```

Reply:
Fair.  I think I actually wrote this one first and inlined the others.

Reply:
Updated